File: learning/pub_sub_cpp/src/pub_sub/src/publisher.cpp
#include <chrono>
#include <functional>
#include <memory>

#include <rclcpp/rclcpp.hpp>
#include <geometry_msgs/msg/twist.hpp>

using namespace std::chrono_literals;

class Publisher : public rclcpp::Node
{
public:
    Publisher() : Node("custom_publisher"), m_Count(0)
    {
        m_Publisher = this->create_publisher<geometry_msgs::msg::Twist>("/turtle1/cmd_vel", 10);
        this->m_Timer = this->create_wall_timer(
            500ms,
            std::bind(&Publisher::publish_vel, this)
        );
    }


private:
    rclcpp::TimerBase::SharedPtr m_Timer;
    rclcpp::Publisher<geometry_msgs::msg::Twist>::SharedPtr m_Publisher;
    size_t m_Count;

private:
    void publish_vel()
    {
        geometry_msgs::msg::Twist message = geometry_msgs::msg::Twist();

        message.linear.x = 1.0;
        message.angular.z = 1.0;
        RCLCPP_INFO(this->get_logger(), "Publishing");
        m_Publisher->publish(message);

    }
};



int main(int argc, char** argv)
{
    rclcpp::init(argc, argv);
    
    std::shared_ptr<Publisher> publisher = std::make_shared<Publisher>(); 

    rclcpp::spin(publisher);
    
    rclcpp::shutdown();
}

File: learning/py_pub_sub_2/src/pub_sub/pub_sub/publisher.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from geometry_msgs.msg import Twist

class Publisher(Node):
    def __init__(self):
        super().__init__("Custom_Publisher")
        self.publish_ = self.create_publisher(Twist, "/turtle1/cmd_vel", 10)
        self.timer_callback = self.create_timer(1.0, self.msg_publisher)

    def msg_publisher(self):
        msg = Twist()
        msg.linear.x = 5.0
        msg.angular.z = 2.0
        self.publish_.publish(msg)
        self.get_logger().info("Publishing")


def main(args = None):
    rclpy.init()
    node = Publisher()
    rclpy.spin(node)
    rclpy.shutdown()